src/ol_infrastructure/applications/bootcamps_ecommerce/__main__.py

```diff
@@ -0,0 +1,203 @@
+                    "Action": "s3:GetObject",
+                    "Resource": f"arn:aws:s3:::{bootcamps_storage_bucket_name}/courses/*",  # noqa: E501
+                }
+            ],
+        }
+    ),
+    tags=aws_config.tags,
+)
+
+
+bootcamps_iam_policy = iam.Policy(
+    f"bootcamps-{stack_info.env_suffix}-policy",
+    description="AWS access controls for the bootcamps application in the "
+    f"{stack_info.name} environment",
+    path=f"/ol-applications/bootcamps/{stack_info.env_suffix}/",
+    name_prefix="aws-permissions-",
+    policy=lint_iam_policy(
+        {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Action": "s3:ListAllMyBuckets",
+                    "Resource": "*",
+                },
+                {
+                    "Effect": "Allow",
+                    "Action": [
+                        "s3:*MultiPartUpload*",
+                        "s3:ListBucket*",
+                        "s3:PutObject*",
+                        "s3:GetObject*",
+                        "s3:DeleteObject*",
+                    ],
+                    "Resource": [
+                        f"arn:aws:s3:::{bootcamps_storage_bucket_name}",
+                        f"arn:aws:s3:::{bootcamps_storage_bucket_name}/*",
+                    ],
+                },
+                {
+                    "Effect": "Allow",
+                    "Action": ["execute-api:Invoke", "execute-api:ManageConnections"],
+                    "Resource": "arn:aws:execute-api:*:*:*",
+                },
+            ],
+        },
+        stringify=True,
+        parliament_config={
+            "PERMISSIONS_MANAGEMENT_ACTIONS": {
+                "ignore_locations": [{"actions": ["s3:putobjectacl"]}]
+            },
+            "RESOURCE_EFFECTIVELY_STAR": {},
+        },
+    ),
+)
+
+bootcamps_vault_backend_role = vault.aws.SecretBackendRole(
+    f"bootcamps-app-{stack_info.env_suffix}",
+    name=f"bootcamps-app-{stack_info.env_suffix}",
+    backend="aws-mitx",
+    credential_type="iam_user",
+    policy_arns=[bootcamps_iam_policy.arn],
+)
+
+# Create RDS instance
+bootcamps_db_security_group = ec2.SecurityGroup(
+    f"bootcamps-db-access-{stack_info.env_suffix}",
+    description=f"Access control for the bootcamps DB in {stack_info.name}",
+    ingress=[
+        ec2.SecurityGroupIngressArgs(
+            protocol="tcp",
+            from_port=DEFAULT_POSTGRES_PORT,
+            to_port=DEFAULT_POSTGRES_PORT,
+            cidr_blocks=["0.0.0.0/0"],
+            ipv6_cidr_blocks=["::/0"],
+            description="Allow access over the public internet from Heroku",
+        ),
+        ec2.SecurityGroupIngressArgs(
+            protocol="tcp",
+            from_port=DEFAULT_POSTGRES_PORT,
+            to_port=DEFAULT_POSTGRES_PORT,
+            security_groups=[data_vpc["security_groups"]["integrator"]],
+        ),
+    ],
+    egress=[
```

Comment:
Should we look into tightening things and if that's the case, is this egress rule necessary? My understanding is that sec groups are stateful and since the Heroku app will be initiating the connection, there's no need for an egress rule. Think it's worth trying out and verifying.